edb/server/compiler/ddl.py

```diff
@@ -207,38 +220,42 @@ def compile_and_apply_ddl_stmt(
 
         current_tx.update_schema(new_schema)
 
-        return dbstate.DDLQuery(
-            sql=(b'SELECT LIMIT 0',),
+        query = dbstate.DDLQuery(
+            sql=NIL_QUERY,
             user_schema=current_tx.get_user_schema(),
             is_transactional=True,
             warnings=tuple(delta.warnings),
             feature_used_metrics=None,
         )
 
+        return query, None
+
     # Apply and adapt delta, build native delta plan, which
     # will also update the schema.
     block, new_types, config_ops = _process_delta(ctx, delta)
 
     ddl_stmt_id: Optional[str] = None
-
     is_transactional = block.is_transactional()
     if not is_transactional:
-        sql = tuple(stmt.encode('utf-8') for stmt in block.get_statements())
+        if not isinstance(stmt, qlast.DatabaseCommand):
+            raise AssertionError(
+                f"unexpected non-transaction DDL command type: {stmt}")
+        sql_stmts = block.get_statements()
+        sql = sql_stmts[0].encode("utf-8")
+        db_op_trailer = tuple(stmt.encode("utf-8") for stmt in sql_stmts[1:])
     else:
-        sql = (block.to_string().encode('utf-8'),)
-
         if new_types:
             # Inject a query returning backend OIDs for the newly
             # created types.
             ddl_stmt_id = str(uuidgen.uuid1mc())
             new_type_ids = [
                 f'{pg_common.quote_literal(tid)}::uuid' for tid in new_types
             ]
-            sql = sql + (
-                trampoline.fixup_query(textwrap.dedent(
-                    f'''\
-                SELECT
-                    json_build_object(
+            new_types_sql = textwrap.dedent(f"""\
```

Comment:
Add a comment about what we're doing here.

Reply:
Added, I also added an explicit `edgedb.indirect_return()` helper to make things even more clear.